### PR TITLE
Explode verification of transparent release endorsements to cover all cases.

### PR DESCRIPTION
**Pull request by thmsbinder**
- Extend endorsements and reference values so that they are in a 1:1 correspondence w.r.t. subfields

### CODE REVIEW

File: oak_attestation_verification/src/verifier.rs [hunk 1]
```diff
@@ -121,43 +122,166 @@ fn verify_dice_chain(evidence: &Evidence) -> anyhow::Result<()> {
...
 
 fn verify_oak_containers(
     now_utc_millis: i64,
-    _evidence: &Evidence,
     endorsements: &OakContainersEndorsements,
     reference_values: &OakContainersReferenceValues,
 ) -> anyhow::Result<()> {
-    let ends_layer = endorsements
+    // Bind references to layers to variables.
+    let e0 = endorsements
```
Comment (k-naliuka) [inline oak_attestation_verification/src/verifier.rs:196]: Would it be possible to use more descriptive variable names, e.g., root_layer_endorsements/ref_values? Or maybe even use the fields directly, especially if you create separate functions for layers (see my next suggestion)?

File: oak_attestation_verification/src/verifier.rs [hunk 2]
```diff
@@ -121,43 +122,166 @@ fn verify_dice_chain(evidence: &Evidence) -> anyhow::Result<()> {
...
 
-    let stage0_end = ends_layer
-        .stage0
+    // Verify Transparent Release endorsements in root layer.
+    verify_root_layer_endorsements(now_utc_millis, e0, r0)?;
+
+    // Verify Transparent Release endorsements in kernel layer.
+    verify_kernel_layer_endorsements(now_utc_millis, e1, r1)?;
+
+    // Verify Transparent Release endorsements in system layer.
```
Comment (k-naliuka) [inline oak_attestation_verification/src/verifier.rs:236]: Should this be extracted into a separate verify_system_layer_endorsements? In general, having a 1:1 correspondence between separate proto messages and these functions will probably make maintenance easier

**Review by k-naliuka — APPROVED** [review]

File: oak_attestation_verification/src/verifier.rs [hunk 3]
```diff
@@ -121,43 +122,166 @@ fn verify_dice_chain(evidence: &Evidence) -> anyhow::Result<()> {
 }
 
 fn verify_oak_restricted_kernel(
-    _evidence: &Evidence,
```
Comment (conradgrobler) [inline oak_attestation_verification/src/verifier.rs:124]: Will this not include the actual evidence later to compare it with the endorsements and reference values?

File: oak_attestation_verification/src/verifier.rs (hunk 1 above)
Comment (conradgrobler) [reply oak_attestation_verification/src/verifier.rs:196]: +1

**Review by conradgrobler — COMMENTED** [review]
Thanks

**Pull request merged**